packages/supplementary-contracts/contracts/blacklist/Blacklist.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+import "./IMinimalBlacklist.sol";
+
+/**
+ * @title Blacklist
+ * @dev List of blacklisted addresses on Ethereum
+ * @dev Source: https://etherscan.io/address/0x97044531D0fD5B84438499A49629488105Dc58e6#readContract
+ */
+contract Blacklist is IMinimalBlacklist, AccessControlEnumerable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @notice We use a mapping in addition to an EnumerableSet to make reading from the storage
+    /// addresses easier
+    // solhint-disable-next-line named-parameters-mapping
+    mapping(address => bool) public isBlacklisted;
+
+    /// @notice We use an EnumerableSet to make lookups and iteration easier
+    // solhint-disable-next-line state-visibility
+    EnumerableSet.AddressSet blacklistSet;
+
+    /// @notice Permissioned role able to add and remove addresses from the blacklist
+    bytes32 public constant UPDATER_ROLE = keccak256("UPDATER_ROLE");
+
+    /// @notice Only Updater modifier
+    modifier onlyUpdater() {
+        // solhint-disable-next-line gas-custom-errors
+        require(hasRole(UPDATER_ROLE, msg.sender), "Must be updater");
+        _;
+    }
+
+    /// @notice Event emitted when an address is added to the blacklist
+    event Blacklisted(address indexed _account);
+
+    /// @notice Event emitted when an address is removed from the blacklist
+    event UnBlacklisted(address indexed _account);
+
+    /**
+     * @notice Construct a new Blacklist contract
+     * @param _admin Blacklist admin
+     * @param _updater Initial updater address
+     * @param _blacklist Initial blacklist
+     */
+    constructor(address _admin, address _updater, address[] memory _blacklist) {
+        _setupRole(UPDATER_ROLE, _updater);
```

Comment:
can roles be transferred?

Reply:
Yup, via OpenZeppelin's `AccessControl` inheritance:

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/dc44c9f1a4c3b10af99492eed84f83ed244203f6/contracts/access/AccessControl.sol#L145